### ISSUE TITLE
coefs fails on interaction effects

### ISSUE THREAD

**Issue by jebyrnes** [jslefche/piecewiseSEM#172]
```
data(meadows)
meadows$grazed <- factor(meadows$grazed)

jutila <- psem(
  lm(rich ~ grazed*(elev + mass), data = meadows),
  lm(mass ~ grazed*elev, data = meadows)
)

coefs(jutila)
```

produces

`Error in `[.data.frame`(data, , vars) : undefined columns selected`

This seems to be due to an element at line 159 in coefs.R.  I'm not 100% sure what you're doing there - could it be accomplished with either removing the interactions or using `model.frame`?

**Comment by jebyrnes** [comment]
Tried the older formulation - janky output - part of which is because of the use of ANOVA with 

```
    levs <- lapply(factorVars, function(j) (emmeans::emmeans(model, list(formula(paste(" ~", j))), data = data)))
```

This gives bad results when an interaction is present, and suppressMessages doesn't warn the user.

**Comment by jebyrnes** [comment]
OK - I fixed it. In general, the whole pulling out factors in getCoefficients was a bad idea, I think. We applied different tests without the user knowing it, and then it lead to weird filtering problems. I like the use of emmeans, but that's something for postdoc explorations by the user. We can build in something that's more dynamic, but, for the moment, let's at least give them meaningful output rather than bad output.
